src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -71,7 +72,9 @@ static wchar_t* toWideCharsArray(const QString& text) {
 
 FILE* BAMUtils::openFile(const QString& fileUrl, const QString& mode) {
 #ifdef Q_OS_WIN
-    QScopedPointer<wchar_t> unicodeFileName(toWideCharsArray(fileUrl));
+    CHECK(!mode.contains('r') || QFileInfo::exists(fileUrl), nullptr);
```

Comment:
Why do we need this check? Looks like unrelated to the main patch

Reply:
Why is the `isBinary` flag needed?

Reply:
> So may be you will want to put it into a separate patch (up to you).

Sounds right. I will do so.

Reply:
✓